tests/functional/Extension/FrontMatterExtension/FrontMatterExtensionTest.php

```diff
@@ -17,16 +21,30 @@
 
 final class FrontMatterExtensionTest extends TestCase
 {
-    private Environment $environment;
+    /**
+     * @return FrontMatterDataParserInterface[]
+     */
+    public function parserProvider(): array
+    {
+        return [
+            [new SymfonyYamlFrontMatterParser()],
+            [new LibYamlFrontMatterParser()],
```

Comment:
The tests that use `ext-yaml` are failing on PHP 8.1 because `ext-yaml` isn't getting installed.  Perhaps we can somehow mark these tests as skipped?

One possible approach could be providing a second variable that tells the test whether it's possible to use that parser:

```php
public function parserProvider(): array
{
            [new SymfonyYamlFrontMatterParser(), true],
            [new LibYamlFrontMatterParser(), LibYamlFrontMatterParser::isCapable()],
}
```

And in the tests we could have something check that variable:

```php
public function testSomething(FrontMatterDataParserInterface $parser, bool $shouldTest): void
{
    $this->skipIfParserNotAvailable($parser, $shouldTest);

    // ... existing test code goes here
}

// ...

private function skipIfParserNotAvailable(FrontMatterDataParserInterface $parser, bool $shouldTest): void
{
    if (! $shouldTest) {
        $this->markTestSkipped(\sprintf('Cannot test with %s due to missing prerequisites', \get_class($parser)));
    }
```

WDYT?

Reply:
Of course, it is better to skip the test if the parser is not available, the modification is pushed.